src/fns.ts

```diff
@@ -1,9 +1,21 @@
-import { size } from 'obj-walker'
 import _debug from 'debug'
-import { Deferred, QueueOptions, QueueResult, WaitOptions } from './types'
+import { sumBy } from 'lodash'
 import makeError from 'make-error'
+import { size } from 'obj-walker'
+
+import {
+  Deferred,
+  QueueOptions,
+  QueueResult,
+  ThroughputLimiterOptions,
+  WaitOptions,
+} from './types'
 
-const debug = _debug('prom-utils')
+const rl = _debug('prom-utils:rateLimit')
+const tl = _debug('prom-utils:throughputLimiter')
+const bq = _debug('prom-utils:batchQueue')
+const pm = _debug('prom-utils:pacemaker')
+const wu = _debug('prom-utils:waitUntil')
```

Comment:
[suggestion] I think it would be cleaner if each of these were isolated in scope to the function they're intended for. For example:

```typescript
export const rateLimit = <T = unknown>(limit: number) => {
  const debug = _debug('prom-utils:rateLimit')
  debug('limit: %d', limit)
  // ...
}
```

Reply:
This requires the function returned by `debug` to be generated for each invocation of `rateLimit`, for example, which seems less ideal than generating it once when the file is loaded.

Reply:
Another option is to do something like this:

```typescript
const rlDebug = _debug('prom-utils:rateLimit')

export const rateLimit = <T = unknown>(limit: number) => {
  const debug = rlDebug
  debug('limit: %d', limit)
  // ...
}
```

Reply:
Or maybe even:

```typescript
const debug = {
  rateLimit: _debug('prom-utils:rateLimit'),
  waitUntil: _debug('prom-utils:waitUntil'),
  // etc.
}

export const rateLimit = <T = unknown>(limit: number) => {
  const debug = debug.rateLimit
  debug('limit: %d', limit)
  // ...
}
```

Reply:
> I see -- good point. Just to avoid confusion when we see those function calls in the code, what do you think about calling these functions e.g. `rlDebug`?

if we go with this, maybe a prefix is better `debugRL` (caps seem better since we are referring to acronyms)

Reply:
I started out with `rlDebug` :). Let me take another pass at this.

Reply:
Changed everything to the `debugRL` naming convention.